client/deposits/list/test/index.js

```diff
@@ -122,4 +135,99 @@ describe( 'Deposits list', () => {
 
 		expect( tableSummary ).toHaveLength( 1 );
 	} );
+
+	describe( 'Download button', () => {
+		test( 'renders when there are one or more deposits', () => {
+			useDeposits.mockReturnValue( {
+				deposits: mockDeposits,
+				isLoading: false,
+			} );
+
+			const { queryByRole } = render( <DepositsList /> );
+			const button = queryByRole( 'button', { name: 'Download' } );
+
+			expect( button ).not.toBeNull();
+		} );
+
+		test( 'does not render when there are no deposits', () => {
+			useDeposits.mockReturnValue( {
+				deposits: [],
+				isLoading: false,
+			} );
+
+			const { queryByRole } = render( <DepositsList /> );
+			const button = queryByRole( 'button', { name: 'Download' } );
+
+			expect( button ).toBeNull();
+		} );
+	} );
+
+	describe( 'CSV download', () => {
+		beforeEach( () => {
+			useDeposits.mockReturnValue( {
+				deposits: mockDeposits,
+				depositsCount: 2,
+				isLoading: false,
+			} );
+			useDepositsSummary.mockReturnValue( {
+				depositsSummary: {
+					count: 2,
+					total: 5000,
+				},
+				isLoading: false,
+			} );
+		} );
+
+		afterEach( () => {
+			jest.resetAllMocks();
+		} );
+
+		afterAll( () => {
+			jest.restoreAllMocks();
+		} );
+
+		test( 'should render expected columns in CSV when the download button is clicked', () => {
+			const { getByRole } = render( <DepositsList /> );
+			getByRole( 'button', { name: 'Download' } ).click();
+
+			const expected = [
+				'',
+				'Date',
+				'Type',
+				'Amount',
+				'Status',
+				'"Bank account"',
+			];
+
+			const csvContent = downloadCSVFile.mock.calls[ 0 ][ 1 ];
+			const csvHeaderRow = csvContent.split( os.EOL )[ 0 ].split( ',' );
+			expect( csvHeaderRow ).toEqual( expected );
+		} );
+
+		test( 'should match the visible rows', () => {
+			const { getByRole, getAllByRole } = render( <DepositsList /> );
+			getByRole( 'button', { name: 'Download' } ).click();
+
+			const csvContent = downloadCSVFile.mock.calls[ 0 ][ 1 ];
+			const csvLines = csvContent.split( os.EOL );
+			const cptRows = getAllByRole( 'row' );
+
+			expect( csvLines.length ).toEqual( cptRows.length );
+
+			const cptFirstDepositDate = cptRows[ 1 ].querySelector( 'td' )
+				.textContent;
+			const csvFirstDeposit = csvLines[ 1 ].split( ',' );
+			const csvFirstDepositDate = csvFirstDeposit[ 1 ];
+			const date = new Date( JSON.parse( csvFirstDepositDate ) );
+			const csvFirstDepositFormattedDate = dateI18n(
+				'M j, Y',
+				moment.utc( date.getTime() ).toISOString(),
+				true // TODO Change call to gmdateI18n and remove this deprecated param once WP 5.4 support ends.
+			);
+
+			expect( csvFirstDepositFormattedDate ).toEqual(
```

Comment:
Just to clarify, here we are testing if the first deposit date is rendered in the table vs the CSV first date?

Reply:
That's true. Besides, I like @htdat's [suggestion](https://github.com/Automattic/woocommerce-payments/pull/2778#discussion_r698270251) of comparing all of the first row's fields rather than just date.